src/core/models/payloads/tagged_data.c

```diff
@@ -0,0 +1,242 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <string.h>
+
+#include "core/models/message.h"
+#include "core/models/payloads/tagged_data.h"
+#include "core/utils/macros.h"
+
+tagged_data_t *tagged_data_new() {
+  tagged_data_t *tagged_data = calloc(1, sizeof(tagged_data_t));
+  if (tagged_data) {
+    tagged_data->tag = byte_buf_new();
+    tagged_data->data = byte_buf_new();
+    if (!tagged_data->tag || !tagged_data->data) {
+      byte_buf_free(tagged_data->tag);
+      byte_buf_free(tagged_data->data);
+      free(tagged_data);
+      return NULL;
+    }
+    return tagged_data;
+  }
+  return NULL;
+}
+
+void tagged_data_free(tagged_data_t *tagged_data) {
+  if (tagged_data) {
+    byte_buf_free(tagged_data->tag);
+    byte_buf_free(tagged_data->data);
+    free(tagged_data);
+  }
+}
+
+tagged_data_t *tagged_data_create(char const *tag, byte_t data[], uint32_t data_len) {
```

Comment:
`char const *tag` indcates it's a pointer to a single character but the tag is an array, suggest change to
```suggestion
tagged_data_t *tagged_data_create(char const tag[], byte_t data[], uint32_t data_len) {
```

Reply:
Done, thanks.